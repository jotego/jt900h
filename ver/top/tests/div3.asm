    ; div RR,r
    maxmode on
    relaxed on
    org 0
    ld a,0xbf    ; common header

    ; 32-bit number divided by 16-bit number
    ; xwa / bc
    ld xwa,354236
    ld bc,1050   ; 0x41a
    div xwa,bc
    jp ov,bad_end
    or ra3,1

    cp wa,337
    jp ne,bad_end
    or ra3,2

    cp qwa0,386
    jp ne,bad_end
    or ra3,4


end_loop:
    ldf 0
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0xcafe,0xbeef,0x1234,0x5678,0x9abc
    end