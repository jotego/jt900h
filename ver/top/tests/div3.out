018201f4 - 00000000 - 00000000 - 0000001f
000002f5 - 00000000 - 00000000 - 00000000
00eb02e9 - 00000000 - 00000000 - 00000000
00ebbabe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - fffffffc
SR = f800
